test/nimble.phx.gen.template/addons/credo_test.exs

```diff
@@ -0,0 +1,55 @@
+defmodule Nimble.Phx.Gen.Template.Addons.CredoTest do
+  use Nimble.Phx.Gen.Template.AddonCase
+
+  setup %{project: project, test_project_path: test_project_path} do
```

Comment:
Just an idea 💡 How about we move this setup to `AddonCase` then make use of tags
```elixir
defmodule Nimble.Phx.Gen.Template.AddonCase do
  # ...

  setup context do
    # ...
    
    project = Nimble.Phx.Gen.Template.Project.info()

    if context[:apply_test_env] do
      File.cd!(test_project_path, fn ->
        Addons.TestEnv.apply(project)
      end)
    end

    {:ok, project: project, test_project_path: test_project_path}
  end
end
```

Then in test, we just add a tag and get rid of `setup` block 
```elixir
describe "#apply/2" do
    @tag :apply_test_env
    test "copies the .credo.exs" do
      # ...
    end

    @tag :apply_test_env
    test "injects credo to mix dependency" do
      # ...
    end
end
```

What do you think? 😄

Reply:
Good suggestion. Note that this `AddonCase` would be define the directory `/test`.

Reply:
@olivierobert I just following Phoenix project as they put the `xxxCase` in the `test/support` folder 
![image](https://user-images.githubusercontent.com/11751745/95168297-49a95b00-07db-11eb-8795-d3cbb939cdbf.png)

Reply:
Yes. I noticed that afterwards and updated my comment above

Reply:
Actually, I prefer to keep the setup here, because that easy to read and know what we need to set up when we check the `credo_test` file, if we go with the tags, we need to find in the source code what and where we handle the `apply_test_env` there.

And think about if we have another case like this, we also put to the AddonCase, that will make the AddonCase bigger.

Another thing, in this test file, we have only 2 test cases, so we can easily put the `@tag :apply_test_env` 2 times, but if we have 10 test cases, we need to put the `@tag :apply_test_env` 10 times ☁️

Reply:
How about this implementation @hanam1ni @olivierobert https://github.com/nimblehq/elixir-templates/pull/26/commits/8a8ec445839df370405291d770c1b078fe74d777

Reply:
looks good to me 🙌 . How about rename from `presetup` to make it less generic and more descriptive,  maybe `base_addons` or `apply_addons` (feel free to use others name 😄 ). 

Approving the PR 👍

Reply:
or `pre_apply` or `required_addons` 🤔  @olivierobert do you have any suggestion 😄

Reply:
Changed it to `required_addons` @hanam1ni 😃